src/services/user.service.js

```diff
@@ -0,0 +1,35 @@
+import bcrypt from "bcrypt";
+import JWT from "jsonwebtoken";
+import { User } from "../database/models/index.js";
+
+export class UserService {
+  static async register(data) {
+    const knownUser = await User.findOne({
+      where: { email: data.email },
+    });
+    if (knownUser) {
+      return "User Exists";
+    } else {
+      const pasSalt = await bcrypt.genSalt(10);
+      const pasHash = await bcrypt.hash(data.password, pasSalt);
+      const newUser = await User.create({
+        firstname: data.firstname,
+        lastname: data.lastname,
+        email: data.email,
+        phone: data.phone,
+        password: pasHash,
+      });
+      await newUser.save();
+      const token = JWT.sign(
+        { id: newUser.id, email: newUser.email },
+        process.env.SEC_TOKEN
+      );
+      return {
+        type: "created",
+        Name: newUser.firstname,
+        Email: newUser.email,
+        data: token,
+      };
+    }
+  }
+}
```

Comment:
Service should hold only queries.
For example: **_When you're creating a new user_**, User service should be
`
export const createUser= async (user) => {
  return await User.create(user);
};
`

Reply:
- Code for hashing passwords should be in helpers
- Code for generating and verifying token should be in helper